lib/Wikia/src/Service/User/Preferences/PreferenceServiceImpl.php

```diff
@@ -235,23 +235,30 @@ private function deleteFromCache( $userId ) {
 	}
 
 	private function applyDefaults( UserPreferences $preferences ) {
-		foreach ( $this->defaultPreferences as $name => $val ) {
-			if ( !$preferences->hasGlobalPreference( $name ) ) {
-				$preferences->setGlobalPreference( $name, $val );
+		foreach ( $this->defaultPreferences->getGlobalPreferences() as $globalPreference ) {
+			if ( !$preferences->hasGlobalPreference( $globalPreference->getName() ) ) {
+				$preferences->setGlobalPreference( $globalPreference->getName(), $globalPreference->getValue() );
+			}
+		}
+
+		foreach ( $this->defaultPreferences->getLocalPreferences() as $wikiId => $localPreferences ) {
+			foreach ( $localPreferences as $localPreference ) {
+				/** @var LocalPreference $localPreference */
+				if ( !$preferences->hasLocalPreference( $localPreference->getName(), $wikiId ) ) {
+					$preferences->setLocalPreference( $localPreference->getName(), $wikiId, $localPreference->getValue() );
+				}
```

Comment:
No. The `PreferenceModule` is iterating over a map of options (`name => val`) and creating a new `UserPreferences` object, this is iterating over a `UserPreferences` object and filling in gaps in the provided `UserPreferences` object